source/base/Shmoose.cpp

```diff
@@ -248,17 +255,30 @@ void Shmoose::sendMessage(QString const &message, QString const &type)
 }
 
 
-void Shmoose::sendFile(QString const &toJid, QString const &file)
+void Shmoose::sendFile(QString toJid, QString file)
 {
     bool shouldEncryptFile = lurchAdapter_->isOmemoUser(toJid) && (! settings_->getSendPlainText().contains(toJid));
+    Swift::JID receiverJid(toJid.toStdString());
+
+    bool success = httpFileUploadManager_->requestToUploadFileForJid(file, toJid, shouldEncryptFile);
+
+    // messsage is added to the database in any case
+    persistence_->addMessage( httpFileUploadManager_->getMsgId(),
+                          QString::fromStdString(receiverJid.toBare().toString()),
+                          QString::fromStdString(receiverJid.getResource()),
+                          file, httpFileUploadManager_->getFileMimeType(), 0, shouldEncryptFile ? 1 : 0);
```

Comment:
This breaks the interface. Even it is possible, don't do direct DB changes here.

Reply:
Ok. Is your suggestion to put it in MessageHandler? I put it here because I saw some calls to persistence in Shmoose.

Reply:
Yes you are right. Shmoose needs the access to the persistence. Only to have the correct functionality in the correct classes, please just move that code to MessageHandler::injectDatabaseEntry (or something like this) and call this here. 

Shmoose.cpp should just be used for the initial connection setup and acts like a central proxy to forward calls to the appropriate classes.